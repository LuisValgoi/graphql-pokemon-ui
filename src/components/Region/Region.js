import React from 'react';

import { useIsSmallScreen } from '../../hooks/useIsMobile';

export const RegionTitle = (props) => {
  const { children, style } = props;
  const region = {
    wrapper: {
      border: '.5rem solid #F7F7F7'
    },
    title: {
      padding: '.5rem',
      color: '#007bff',
      border: '1px solid rgba(0,0,0,.125)',
      backgroundColor: 'white',
      borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px)',
      verticalAlign: 'middle',
      textAlign: 'center'
    }
  }

  return (
    <div style={{ ...region.wrapper, ...style }}>
      <div style={region.title}>
        {children}
      </div>
    </div >
  );
};

export const RegionImage = (props) => {
  const { children, style } = props;
  const region = {
    wrapper: {
      border: '.5rem solid #F7F7F7',
      backgroundColor: 'white',
      verticalAlign: 'middle',
      textAlign: 'center'
    },
    image: {
      border: '1px solid rgba(0,0,0,.125)',
      borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px)'
    }
  };

  return (
    <div style={{ ...region.wrapper, ...style }}>
      <div style={region.image}>
        {children}
      </div>
    </div>
  );
}

export const RegionContent = (props) => {
  const { children, style } = props;
  const region = {
    wrapper: {
      border: '.5rem solid #F7F7F7'
    },
    info: {
      border: '1px solid rgba(0,0,0,.125)',
      backgroundColor: 'white',
      borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px)',
      padding: '.5rem'
    }
  }

  return (
    <div style={{ ...region.wrapper, ...style }}>
      <div style={region.info}>
        {children}
      </div>
    </div >
  );
};

export const Region = (props) => {
  const isSmallScreen = useIsSmallScreen();
  const { children } = props;
  const region = {
    wrapper: {
      marginTop: isSmallScreen ? '3rem' : '1rem',
      padding: '.1rem',
      border: '1px solid rgba(0,0,0,.125)',
      borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px)',
      backgroundColor: '#F7F7F7'
    }
  };
  return (
    <div style={region.wrapper}>
      {children}
    </div>
  )
};
