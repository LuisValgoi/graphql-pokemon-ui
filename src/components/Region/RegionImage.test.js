import React from 'react';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { RegionImage } from './Region';

afterEach(cleanup);

test('RegionImage: should render', () => {
  const { asFragment } = render(<RegionImage />)
  expect(asFragment()).toMatchSnapshot();
});

test('RegionImage: should render img w/ some borders around it', () => {
  const img = React.createElement('img', { src: 'https://via.placeholder.com/150' });
  const { getByTestId } = render(<RegionImage>{img}</RegionImage>);
  const wrapper = getByTestId('region-image-wrapper');
  const info = getByTestId('region-image-info');
  expect(wrapper).toContainElement(info);
  expect(info).toHaveStyle(`border: '1px solid rgba(0,0,0,.125)'`);
});
