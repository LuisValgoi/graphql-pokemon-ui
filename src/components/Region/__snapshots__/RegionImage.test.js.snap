// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionImage: should render 1`] = `
<DocumentFragment>
  <div
    data-testid="region-image-wrapper"
    style="border: .5rem solid #f7f7f7; background-color: white; vertical-align: middle; text-align: center;"
  >
    <div
      data-testid="region-image-info"
      style="border-radius: calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px) calc(.25rem - 1px);"
    />
  </div>
</DocumentFragment>
`;
